stdlib/public/core/CharacterProperties.swift

```diff
@@ -23,11 +23,19 @@ extension Character {
   }
 
   /// A Boolean value indicating whether this is an ASCII character.
-  @inlinable
+  @_alwaysEmitIntoClient @_silgen_name("inlineIsASCII")
   public var isASCII: Bool {
-    return asciiValue != nil
+    if !_str._guts.isSmall { return outlineIsASCII }
+    // This character is backed by a small string. We can check if it is a
+    // single ASCII character (and return its value) just by looking at the
+    // first word. This works because Character is made up of a single
+    // grapheme cluster, the NUL character does not combine, and in a
+    // SmallString all unused bytes must be zero, so we don't need to look
+    // at the count.
+    let firstWord = _str._guts.asSmall._storage.0.littleEndian
```

Comment:
I could be mistaken, but I think this is the first place in the stdlib where _SmallString is not behind a resilience boundary. I'm not going to say that's wrong, but it does seem like something we should consider carefully and check with @milseman on.